### DISCUSSION TITLE
How to correctly use AutoCompleteBox

### DISCUSSION THREAD

**Discussion by JaoHundred** [AvaloniaUI/Avalonia#13070] [Q&A] [answered]
Hi, I'm trying to wrap my head around this control. The documentation and examples for this are scarce to complex scenarios as list of complex objects. I though it was similar in use with ComboBox but I doesn't found any equivalent property for DisplayMemberBinding. Could someone point me in the right direction?

**Comment by timunie** [comment, marked as answer]
simplest idea is probably to override `ToString` in your ItemViewModel.

**Reply by JaoHundred** [reply]
Good idea, thanks Timunie. Do Avalonia team has plans to update AutoCompleteBox in future to make it more easier to use?

**Reply by dbriard** [reply]
I have difficulty to understand it too, is it supposed to be used for search scenarios? and equivalent to WinUI/UWP AutoSuggestBox?

**Reply by JaoHundred** [reply]
@dbriard , I'm not familiar with uwp AutoSuggestBox. About the uses of AutoCompleteBox, normally you could use it when you need the user to filter larger sets of data. You can see AutoCompleteBox as ComboBox + text filter.

**Reply by dbriard** [reply]
Thanks @JaoHundred 
I see, so the data must be known in advance, and a use case could be for a FontComboBox.
While AutoSuggestBox can be use to search for dynamic async data, such as finding places by providing the address in a maps app for example.

**Reply by timunie** [reply]
AutoCompleteBox has also an AsyncPopulator. Check ControlCatalog for sample: 
https://github.com/AvaloniaUI/Avalonia/blob/39862069aebe8898deb5210913137adb4611697a/samples/ControlCatalog/Pages/AutoCompleteBoxPage.xaml.cs#L151-L158

**Comment by EsztTelek** [comment]
Hey, i do have a problem with it too, i have been trying to make this control work the problem is after i click on one of the suggested word there is a null reference exception on a thread that i cannot debug. I've used the example code (https://docs.avaloniaui.net/docs/next/reference/controls/autocompletebox)
the only difference is that there isn't a property named "Items" (it seems to be private, maybe the sample is outdated) so i've used "ItemsSource" instead. There are no other controls and there is only one line of code that sets the ItemsSource. What did i miss?

CodeBehind:
public List<string> dataz = new List<string>() {"dog", "cat", "lobster"};
...
ACompBox.ItemsSource = dataz;
...

MainView.axaml:
...
`<AutoCompleteBox` Name="ACompBox" FilterMode="StartsWith" />
...

**Reply by timunie** [reply]
Normally, I'd say better to use the MVVM pattern instead of code behind. And populate it via Bindings.

**Reply by EsztTelek** [reply]
Thanks for the fast answer! Yeah that was my first approach but it didn't work so i've figured with code behind there is 1 less thing that could go wrong. I've updated the NuGet packages and now it works like a charm. Thanks for your time, have a nice day! :)
